influxdb/content.md

```diff
@@ -6,59 +6,409 @@ InfluxDB is a time series database built from the ground up to handle high write
+      -e INFLUXD_INIT_USERNAME=my-user \
+      -e INFLUXD_INIT_PASSWORD=my-password \
+      -e INFLUXD_INIT_ORG=my-org \
+      -e INFLUXD_INIT_BUCKET=my-bucket \
+      %%IMAGE%%
+```
+
+#### Upgrade Example - Custom Paths
+
+Assume you've been running a 1.x deployment with data and config mounted at custom paths:
+
+```console
+$ docker run -p 8086:8086 \
+      -v influxdb:/root/influxdb/data \
+      -v $PWD/influxdb.conf:/root/influxdb/influxdb.conf:ro \
+      %%IMAGE%%:1.8 -config /root/influxdb/influxdb.conf
+```
+
+To upgrade this deployment to 2.x, first decide if you'd like to keep using custom paths, or use the 2.x defaults. If you decide to use the defaults, you'd stop the running 1.x container, then run:
+
+```console
+$ docker run -p 8086:8086 \
+      -v influxdb:/root/influxdb/data \
+      -v influxdb2:/var/lib/influxdb2 \
+      -v $PWD/influxdb.conf:/root/influxdb/influxdb.conf:ro \
+      -e INFLUXD_INIT_MODE=upgrade \
+      -e INFLUXD_INIT_USERNAME=my-user \
+      -e INFLUXD_INIT_PASSWORD=my-password \
+      -e INFLUXD_INIT_ORG=my-org \
+      -e INFLUXD_INIT_BUCKET=my-bucket \
+      -e INFLUXDB_INIT_UPGRADE_V1_CONFIG=/root/influxdb/influxdb.conf \
       %%IMAGE%%
 ```
 
+To retain your custom paths, you'd run:
+
+```console
+$ docker run -p 8086:8086 \
+      -v influxdb:/root/influxdb/data \
+      -v influxdb2:/root/influxdb2/data \
+      -v $PWD/influxdb.conf:/root/influxdb/influxdb.conf:ro \
+      -e INFLUXD_INIT_MODE=upgrade \
+      -e INFLUXD_INIT_USERNAME=my-user \
+      -e INFLUXD_INIT_PASSWORD=my-password \
+      -e INFLUXD_INIT_ORG=my-org \
+      -e INFLUXD_INIT_BUCKET=my-bucket \
+      -e INFLUXDB_INIT_UPGRADE_V1_CONFIG=/root/influxdb/influxdb.conf \
+      -e INFLUXD_CONFIG_PATH=/root/influxdb2/config.toml \
+      -e INFLUXD_BOLT_PATH=/root/influxdb2/influxdb.bolt \
+      -e INFLUXD_ENGINE_PATH=/root/influxdb2/engine \
+      %%IMAGE%%
+```
+
+### Custom Initialization Scripts
+
+The InfluxDB image supports running arbitrary initialization scripts after initial system setup, on both the `setup` and `upgrade` paths. Scripts must have extension `.sh` and be mounted inside of the `/docker-entrypoint-initdb.d` directory. When multiple scripts are present, they will be executed in lexical sort order by name.
+
+As a convenience for script-writers, the image will export a number of variables into the environment before executing any scripts:
+* `INFLUX_CONFIGS_PATH`: Path to the CLI configs file written by `setup`/`upgrade`
+* `INFLUX_HOST`: URL to the `influxd` instance running setup logic
+* `INFLUXDB_INIT_USER_ID`: ID of the initial admin user created by `setup`/`upgrade`
+* `INFLUXDB_INIT_ORG_ID`: ID of the initial organization created by `setup`/`upgrade`
+* `INFLUXDB_INIT_BUCKET_ID`: ID of the initial bucket created by `setup`/`upgrade`
+
+For example, if you wanted to grant write-access to a 1.x client on your initial bucket, you'd first create the file `$PWD/scripts/setup-v1.sh` with contents:
```

Comment:
Could be useful to include read and write as first example to reach a larger audience. And show how to use it? Or link to documentation.

Reply:
Added a small section after the info on how to set up a fresh DB, mostly links to the public docs.